agent/src/agent/streamsets/manager.py

```diff
@@ -29,23 +27,17 @@ def delete_streamsets(streamsets_: StreamSets):
                 'Cannot move pipelines to a different streamsets as only one streamsets instance exists, cannot delete streamsets that has pipelines'
             )
         streamsets.manager.StreamsetsBalancer().unload_streamsets(streamsets_)
-    pipeline.manager.delete_monitoring_pipeline(streamsets_)
     streamsets.repository.delete(streamsets_)
 
 
 def _has_pipelines(streamsets_: StreamSets) -> bool:
-    pipelines = pipeline.repository.get_by_streamsets_id(streamsets_.id)
-    if len(pipelines) > 1:
-        return True
-    if len(pipelines) == 1 and not pipeline.manager.is_monitoring(pipelines[0]):
-        return True
-    return False
+    return True if len(pipeline.repository.get_by_streamsets_id(streamsets_.id)) >= 1 else False
```

Comment:
`return len(pipeline.repository.get_by_streamsets_id(streamsets_.id)) > 0`

Reply:
why does it matter?

Reply:
it's just easier to read